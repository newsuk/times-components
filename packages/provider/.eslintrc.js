module.exports = {
  extends: ["@times-components/thetimes"],
  parser: "babel-eslint",
  rules: {
    "graphql/template-strings": [
      "error",
      {
        env: "apollo",
        schemaJson: require("@times-components/schema/schema.json")
      }
    ]
  },
  plugins: ["graphql"]
};
