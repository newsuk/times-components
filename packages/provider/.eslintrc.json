{
  "extends": ["airbnb", "prettier", "prettier/react"],
  "parser": "babel-eslint",
  "rules": {
    "react/jsx-filename-extension": [1, { "extensions": [".js"] }],
    "import/extensions": "off"
  },
  "plugins": ["graphql"],
  "settings": {
    "import/resolver": {
      "node": {
        "extensions": [".js", ".android.js", ".ios.js", ".web.js"]
      }
    }
  },
  "root": true
}
