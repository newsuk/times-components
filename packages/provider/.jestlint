{
  "maxFileSize": 18587,
  "maxLines": 422
}

