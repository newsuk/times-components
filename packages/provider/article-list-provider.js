import gql from "graphql-tag";
import connectGraphql from "./connect";

export const query = gql`
  query ArticleListQuery(
    $slug: Slug!
    $first: Int
    $skip: Int
    $imageRatio: Ratio!
  ) {
    author(slug: $slug) {
      articles {
        count
        list(first: $first, skip: $skip) {
          content(maxCharCount: 145, markupType: "paragraph")
          id
          label
          leadAsset {
            ... on Image {
              title
              crop(ratio: $imageRatio) {
                url
              }
            }
            ... on Video {
              posterImage {
                title
                crop(ratio: $imageRatio) {
                  url
                }
              }
            }
          }
          publicationName
          publishedTime
          title
          url
        }
      }
    }
  }
`;

const propsToVariables = ({
  slug,
  pageSize,
  page,
  articleImageRatio = "3:2"
}) => ({
  slug,
  first: pageSize,
  skip: pageSize * (page - 1),
  imageRatio: articleImageRatio
});

export default connectGraphql(query, propsToVariables);
