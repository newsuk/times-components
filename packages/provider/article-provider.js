import gql from "graphql-tag";
import connectGraphql from "./connect";

export const query = gql`
  query ArticleQuery($id: ID!) {
    article(id: $id) {
      id
      title
      publicationName
      publishedTime
      label
      standfirst
      flags
      byline
      content
      leadAsset {
        ... on Image {
          id
          title
          credits
          caption
          crop(ratio: "16:9") {
            ratio
            url
          }
        }
      }
    }
  }
`;

const propsToVariables = id => id;

export default connectGraphql(query, propsToVariables);
