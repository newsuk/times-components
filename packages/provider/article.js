import gql from "graphql-tag";
import connectGraphql from "./connect";

export const query = gql`
  query ArticleQuery($id: ID!) {
    article(id: $id) {
      id
      headline
      publicationName
      publishedTime
      label
      standfirst
      flags
      byline
      content
      leadAsset {
        ... on Image {
          id
          title
          credits
          caption
          crop(ratio: "16:9") {
            ratio
            url
          }
        }
      }
    }
  }
`;

export default connectGraphql(query);
