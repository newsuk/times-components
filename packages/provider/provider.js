import connect from "./connect";
import AuthorProfileProvider from "./author-profile-provider";
import ArticleProvider from "./article-provider";

export default connect;

export { AuthorProfileProvider, ArticleProvider };
