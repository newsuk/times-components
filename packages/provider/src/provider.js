import connect, { QueryProvider } from "./connect";
import AuthorProfileProvider from "./author-profile";
import AuthorArticlesNoImagesProvider from "./author-articles-no-images";
import AuthorArticlesWithImagesProvider from "./author-articles-with-images";
import ArticleProvider from "./article";
import ArticleCommentsProvider from "./article-comments";
import EditionProvider from "./edition";
import TopicProvider from "./topic";
import TopicArticlesProvider from "./topic-articles";

export default connect;

export {
  AuthorProfileProvider,
  AuthorArticlesNoImagesProvider,
  AuthorArticlesWithImagesProvider,
  ArticleProvider,
  ArticleCommentsProvider,
  EditionProvider,
  TopicProvider,
  TopicArticlesProvider,
  QueryProvider
};
