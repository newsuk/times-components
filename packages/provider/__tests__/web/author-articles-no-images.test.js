import React from "react";
import renderer from "react-test-renderer";
import {
  authorProfile as makeAuthorParams,
  MockedProvider,
  MockFixture
} from "@times-components/provider-test-tools";
import { authorArticlesNoImages as authorArticlesNoImagesQuery } from "@times-components/provider-queries";
import { AuthorArticlesNoImagesProvider } from "../../src/provider";

const renderComponent = child => {
  const pageSize = 1;
  const slug = "deborah-haynes";

  return renderer.create(
    <MockFixture
      params={makeAuthorParams({
        articleQuery: authorArticlesNoImagesQuery,
        articleVariables: iteration => ({
          first: pageSize,
          longSummaryLength: 360,
          shortSummaryLength: 220,
          skip: (iteration - 1) * pageSize,
          slug
        }),
        hasLeadAssets: false,
        pageSize,
        slug
      })}
      render={mocks => (
        <MockedProvider mocks={mocks}>
          <AuthorArticlesNoImagesProvider
            debounceTimeMs={0}
            page={1}
            pageSize={pageSize}
            slug={slug}
          >
            {child}
          </AuthorArticlesNoImagesProvider>
        </MockedProvider>
      )}
    />
  );
};

describe("AuthorArticlesNoImagesProvider", () => {
  it("returns query result", done => {
    renderComponent(({ isLoading, author }) => {
      if (!isLoading) {
        expect(author).toMatchSnapshot();
        done();
      }

      return null;
    });
  });
});
