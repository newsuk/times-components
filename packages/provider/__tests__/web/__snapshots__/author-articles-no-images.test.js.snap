// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorArticlesNoImagesProvider returns query result 1`] = `
Object {
  "__typename": "Author",
  "articles": Object {
    "__typename": "AuthorArticles",
    "count": 200,
    "list": Array [
      Object {
        "__typename": "Article",
        "hasVideo": false,
        "headline": "Test Headline 1",
        "id": "d98c2 1c-cb16-11e7-b529-95e3fc05f40f",
        "label": "Test Label 1",
        "leadAsset": Object {
          "__typename": "Image",
          Symbol(id): "$Article:d98c2 1c-cb16-11e7-b529-95e3fc05f40f.leadAsset",
        },
        "longSummary": Array [],
        "publicationName": "TIMES",
        "publishedTime": "2018-06-01",
        "shortHeadline": "Test Short Headline 1",
        "shortIdentifier": "968n7tdck1",
        "shortSummary": Array [],
        "slug": "this-is-slug-1",
        "url": "https://url.io",
        Symbol(id): "Article:d98c2 1c-cb16-11e7-b529-95e3fc05f40f",
      },
    ],
    Symbol(id): "$ROOT_QUERY.author({\\"slug\\":\\"deborah-haynes\\"}).articles",
  },
  Symbol(id): "$ROOT_QUERY.author({\\"slug\\":\\"deborah-haynes\\"})",
}
`;
